quaddtype/numpy_quaddtype/src/casts.cpp

```diff
@@ -605,6 +605,175 @@ quad_to_bytes_loop(PyArrayMethod_Context *context, char *const data[],
     return 0;
 }
 
+// StringDType to QuadDType casting
+static NPY_CASTING
+stringdtype_to_quad_resolve_descriptors(PyObject *NPY_UNUSED(self), PyArray_DTypeMeta *dtypes[2],
+                                        PyArray_Descr *given_descrs[2], PyArray_Descr *loop_descrs[2],
+                                        npy_intp *view_offset)
+{
+    Py_INCREF(given_descrs[0]);
+    loop_descrs[0] = given_descrs[0];
+
+    if (given_descrs[1] == NULL) {
+        loop_descrs[1] = (PyArray_Descr *)new_quaddtype_instance(BACKEND_SLEEF);
+        if (loop_descrs[1] == nullptr) {
+            Py_DECREF(loop_descrs[0]);
+            return (NPY_CASTING)-1;
+        }
+    }
+    else {
+        Py_INCREF(given_descrs[1]);
+        loop_descrs[1] = given_descrs[1];
+    }
+
+    // no notion of fix length, so always unsafe
+    return NPY_UNSAFE_CASTING;
+}
+
+// Note: StringDType elements are always aligned, so Aligned template parameter
+// is kept for API consistency but both versions use the same logic
+template <bool Aligned>
+static int
+stringdtype_to_quad_strided_loop(PyArrayMethod_Context *context, char *const data[],
+                                 npy_intp const dimensions[], npy_intp const strides[],
+                                 void *NPY_UNUSED(auxdata))
+{
+    npy_intp N = dimensions[0];
+    char *in_ptr = data[0];
+    char *out_ptr = data[1];
+    npy_intp in_stride = strides[0];
+    npy_intp out_stride = strides[1];
+
+    PyArray_Descr *const *descrs = context->descriptors;
+    PyArray_StringDTypeObject *str_descr = (PyArray_StringDTypeObject *)descrs[0];
+    QuadPrecDTypeObject *descr_out = (QuadPrecDTypeObject *)descrs[1];
+    QuadBackendType backend = descr_out->backend;
+
+    npy_string_allocator *allocator = NpyString_acquire_allocator(str_descr);
+
+    while (N--) {
+        const npy_packed_static_string *ps = (npy_packed_static_string *)in_ptr;
+        npy_static_string s = {0, NULL};
+        int is_null = NpyString_load(allocator, ps, &s);
+        
+        if (is_null == -1) {
+            NpyString_release_allocator(allocator);
+            PyErr_SetString(PyExc_MemoryError, "Failed to load string in StringDType to Quad cast");
+            return -1;
+        }
+        else if (is_null) {
+            // Handle null string - use the default string if available, otherwise error
+            if (str_descr->has_string_na || str_descr->default_string.buf != NULL) {
+                s = str_descr->default_string;
+            }
+            else {
+                NpyString_release_allocator(allocator);
+                PyErr_SetString(PyExc_ValueError, "Cannot convert null string to QuadPrecision");
+                return -1;
+            }
+        }
+
+        quad_value out_val;
+        if (bytes_to_quad_convert(s.buf, s.size, backend, &out_val) < 0) {
```

Comment:
utilising `bytes_to_quad_convert` instead of `unicode_to_quad_convert` because the later expects `Py_UCS` format otherwise both are doing the same thing